segmentation/include/pcl/segmentation/impl/extract_clusters.hpp

```diff
@@ -61,21 +62,21 @@ pcl::extractEuclideanClusters (const PointCloud<PointT> &cloud,
   std::vector<int> nn_indices;
   std::vector<float> nn_distances;
   // Process all points in the indices vector
-  for (int i = 0; i < static_cast<int> (cloud.points.size ()); ++i)
-  {
+  auto lambda = [&](int &i) {
     if (processed[i])
-      continue;
+      return;
 
-    std::vector<int> seed_queue;
     int sq_idx = 0;
-    seed_queue.push_back (i);
+    clusters.emplace_back();
+    auto seed_queue = clusters.back();
```

Comment:
Request an explicit reference.  I still think you could create `seed_queue` outside and then emplace+move it into `clusters` if it fulfills the conditions with respect to the number of points.